### ISSUE TITLE
False positive: global-variable-not-assigned with class

### ISSUE THREAD

**Issue by henryiii** [PyCQA/pylint#5756]
### Bug description

If you use `global` with a `class`, this incorrectly thinks the global variable is never assigned. For example, a MWE:

```python

MyClass = None

def register():
    import foo

    global MyClass

    class MyClass(foo.bar):
        pass
```

### Command used

```shell
pylint tmp.py
```

### Pylint output

```shell
tmp.py:6:4: W0602: Using global for 'MyClass' but no assignment is done (global-variable-not-assigned)

-----------------------------------
Your code has been rated at 8.33/10
```

### Expected behavior

I should get a warning about using a global, not about not assigning it. Class definitions are assignments. (also function definitions, too)

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.1 (main, Dec  6 2021, 22:25:40) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

macOS

**Comment by DanielNoord** [comment]
I think this might have been fixed with #5690. @mbyrnepr2 can you reproduce this on `main`?

**Comment by henryiii** [comment]
```console
# pylint tmp.py
************* Module tmp
tmp.py:6:4: W0602: Using global for 'MyClass' but no assignment is done (global-variable-not-assigned)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)

```

```console
$ pipx run --spec git+https://github.com/pycqa/pylint pylint tmp.py                                                                     1247ms  Tue Feb  1 12:50:04 2022
⚠️  pylint is already on your PATH and installed at /usr/local/bin/pylint. Downloading and running anyway.

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.33/10, +1.67)
```

Looks like it! Thanks!

**Comment by DanielNoord** [comment]
🎉

**Comment by mbyrnepr2** [comment]
Great! Glad it worked out!
